src/main/resources/static/js/views.js

```diff
@@ -0,0 +1,116 @@
+'use strict'
+
+
+
+$(document).ready(() => {
+
+//    $('#logo_frame').on('click', () => {
+//        alert('jq works!')
+//    });
+
+    $('.resort_button_div').on('click', getResortSelection);
+
+    function getResortSelection(e){
+
+        e.stopPropagation();
+//        console.log($(this)[0].dataset.id);
+        $.ajax({
+           url:"/resorts/" + $(this)[0].dataset.id,
+           type:"GET"
+        }).done(resort => {
+            console.log(resort);
+            $('#ajax_test').text(resort.name);
+
+            // inserting modMap function to change map center point via map input
+//            let latty = resort.lat;
+//            let longy = resort.long;
+
+            console.log("This is our resort latitude", resort.latitude);
+            console.log("This is our resort longitude", resort.longitude);
+
+            localStorage.setItem('storedLat', JSON.stringify(resort.latitude));
+            localStorage.setItem('storedLong', JSON.stringify(resort.longitude));
+
+            modMap();
+
+  
+            $('#report_container > iframe').attr('src', 'https://www.onthesnow.com/widget/snow?resort=' + resort.widgetId + '&color=b');
+            $('#report_container > p > a').attr('href', resort.otsUrl);
+            $('#teams').empty();
+
+            $('.ots-widget > iframe').attr('src', 'https://www.onthesnow.com/widget/snow?resort=' + resort.widgetId + '&color=b');
+            $('.ots-widget > p > a').attr('href', resort.otsUrl);
+
+                $('#teams').empty();
+
+
+            resort.teams.forEach(team => {
+                let div = $('<div class="team_card" data-id="' + team.id + '"></div>');
+                div.append('<span>' + team.difficulty + ' </span><br>');
+                div.append('<span>' + team.name + ' </span><br>');
+                div.append('<span>' + team.users.length + '/' + team.capacity + ' </span><br>');
+                div.append('<span>' + team.description + ' </span><br>');
+                div.append('<a href="/teams/' + team.id + '">View</a>')
+                $('#teams').append(div);
+
+            });
+
+            $('body').append('<script src="https://maps.googleapis.com/maps/api/js?key=AIzaSyDxgddLplSZhq5H2eEIxCPdacE-VmAWHk0&callback=modMap" async defer></script>');
+
+
+        })
+
+    }
+
+    console.log("in views.js");
+    var map;
+
+    function modMap() {
+
+        var resortLat = JSON.parse(localStorage.getItem('storedLat'));
+        var resortLong = JSON.parse(localStorage.getItem('storedLong'));
+        var myLatlng = new google.maps.LatLng(resortLat, resortLong);
+
+        var mapOptions = {
+          zoom: 9,
+          center: myLatlng,
+          mapTypeId: 'terrain'
+        };
+
+        var location = {lat: resortLat, lng: resortLong};
+
+        map = new google.maps.Map(document.getElementById('map'), mapOptions);
```

Comment:
noooooo noooooooooooooo

according to [this StackOverflow](https://stackoverflow.com/questions/12554688/how-to-dynamically-change-the-centre-of-a-google-maps-without-erasing-markers), you should be able to call `map.setCenter` here instead of having to completely change out the map.

Reply:
Got curious enough to actually figure out the fix. Since this method is called both for initializing a new map, you actually need a tiny conditional:
```
if (map) {
            map.setCenter(myLatlng);
        } else {
            map = new google.maps.Map(document.getElementById('map'), mapOptions);
        }
        new google.maps.Marker({position: location, map: map});
```
And also to get rid of the line that adds a new script tag.